controllers/cronset_controller.go

```diff
@@ -73,8 +78,34 @@ func (r *CronSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *CronSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
+	if err := ctrl.NewControllerManagedBy(mgr).
 		For(&batchv1alpha1.CronSet{}).
 		Owns(&batchv1.CronJob{}).
-		Complete(r)
+		Complete(r); err != nil {
+		return err
+	}
+
+	if err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Node{}).
+		Watches(&source.Kind{Type: &corev1.Node{}},
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				var cronSetObjs batchv1alpha1.CronSetList
+				_ = mgr.GetClient().List(context.TODO(), &cronSetObjs)
+
+				var requests []reconcile.Request
+				for _, obj := range cronSetObjs.Items {
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name:      obj.Name,
+							Namespace: obj.Namespace,
+						},
+					})
+				}
+
+				return requests
```

Comment:
Yes, with the current logic, `Reconcile` will be called as many times as there are CronSets because I didn't add any conditions.
hm.. I think it would be better if it filtered by something like node selector.

Reply:
Do you plan to check all node lists in reconcile?
well.. I thought we should make one request as cronset of node. cause, we could have the rule to make cronjob res name using node.

Reply:
yes. I agree with you. that's cool to do at next step. :)
um. Is there any way that we don't have to check all node list in reconcile func?

I think one of the ways, If we can check the kind of node event and cronset has needed node list, how about it? couldn't we?

Reply:
The values of the ctrl.Request parameter can be distinguished as follows:

- For node events: /node1
- For cronset events: default/test-cronset

Because the namespace value does not exist for node events, we can use this difference to handle them differently than cronset events within reconcile logic.

However, if it's a cronset event, we have to specify in the pod spec which node it's deployed to... so I think that's where we need to check all the nodes.

Reply:
Ah, so instead of checking all the nodes,
we can just query based on a value like node selector and check those nodes.

Reply:
Nit: when cronset is created, and it doesn't have desired node, we should check all nodes, but after configured desired node, we could follow node events. It that not efficient?